src/main/java/org/candlepin/async/tasks/CertificateCleanupJob.java

```diff
@@ -0,0 +1,132 @@
+import org.candlepin.model.ExpiredCertificate;
+import org.candlepin.model.IdentityCertificateCurator;
+
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+/**
+ * The job will periodically delete all expired identity and content certificates
+ * as well as all revoked and expired certificate serials.
+ */
+public class CertificateCleanupJob implements AsyncJob {
+    private static final Logger log = LoggerFactory.getLogger(CertificateCleanupJob.class);
+
+    public static final String JOB_KEY = "CertificateCleanupJob";
+    public static final String JOB_NAME = "Certificate Cleanup";
+    public static final String DEFAULT_SCHEDULE = "0 0 12 * * ?";
+
+    private final Configuration config;
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identCertCurator;
+    private final ContentAccessCertificateCurator caCertCurator;
+    private final CertificateSerialCurator serialCurator;
+
+    @Inject
+    public CertificateCleanupJob(
+        Configuration config,
+        ConsumerCurator consumers,
+        IdentityCertificateCurator identCerts,
+        ContentAccessCertificateCurator contentAccessCerts,
+        CertificateSerialCurator serialCurator) {
+        this.config = Objects.requireNonNull(config);
+        this.consumerCurator = Objects.requireNonNull(consumers);
+        this.identCertCurator = Objects.requireNonNull(identCerts);
+        this.caCertCurator = Objects.requireNonNull(contentAccessCerts);
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+    }
+
+    @Override
+    @Transactional
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        log.debug("Starting certificate cleanup.");
+        // Skipping entitlement certificates as those are handled by ExpiredPoolsCleanupJob
+        cleanupExpiredIdentityCerts();
+        cleanupExpiredContentAccessCerts();
+        cleanupCertificateSerials();
+        log.debug("Certificate cleanup successfully completed.");
+    }
+
+    private void cleanupExpiredIdentityCerts() {
+        List<ExpiredCertificate> allExpiredIdCertificates = this.identCertCurator.listAllExpired();
+        log.info("Cleaning up {} expired identity certificates.", allExpiredIdCertificates.size());
+        for (List<ExpiredCertificate> expiredCertificates : partition(allExpiredIdCertificates)) {
+            List<String> expiredCertIds = certIdsOf(expiredCertificates);
+            List<Long> expiredSerials = serialsOf(expiredCertificates);
+            List<String> consumersToUnlink = this.consumerCurator.findUuidsByIdentCerts(expiredCertIds);
+            int unlinkedConsumers = this.consumerCurator.unlinkIdCertificatesOf(consumersToUnlink);
```

Comment:
Doing these 2 steps as two separate queries runs the risk of a race condition, where if a consumer with an expired cert checks-in in between those 2 queries and gets a new valid certificate, then the `unlinkIdCertificatesOf` method would be unlinking that new valid certificate. Doing it in a single query would make sure the database does it as a single atomic operation

Reply:
In case this is hard to do using HQL, we can always use pure SQL (and make sure to test it against both mariadb + postgres).

Reply:
Done.